wp-content/themes/core/components/content-block/Content_Block.php

```diff
@@ -9,45 +9,41 @@
  * @property string[] $classes
  * @property string[] $attrs
  * @property string[] $container_classes
- * @property string   $leadin
- * @property string   $title
- * @property string   $text
- * @property string   $action
+ * @property array    $leadin
+ * @property array    $title
+ * @property array    $text
+ * @property array    $action
  */
-class Content_Block extends Context {
-	public const TAG               = 'tag';
-	public const CLASSES           = 'classes';
-	public const ATTRS             = 'attrs';
-	public const LEADIN            = 'leadin';
-	public const TITLE             = 'title';
-	public const TEXT              = 'text';
-	public const ACTION            = 'action';
+class Content_Block extends Component {
 
-	protected $path = __DIR__ . '/content-block.twig';
+	public const TAG     = 'tag';
+	public const CLASSES = 'classes';
+	public const ATTRS   = 'attrs';
+	public const LEADIN  = 'leadin';
+	public const TITLE   = 'title';
+	public const TEXT    = 'text';
+	public const ACTION  = 'action';
 
-	protected $properties = [
-		self::TAG               => [
-			self::DEFAULT => 'div',
-		],
-		self::CLASSES           => [
-			self::DEFAULT       => [ 'c-content-block' ],
-			self::MERGE_CLASSES => [],
-		],
-		self::ATTRS             => [
-			self::DEFAULT          => [],
-			self::MERGE_ATTRIBUTES => [],
-		],
-		self::LEADIN            => [
-			self::DEFAULT => '',
-		],
-		self::TITLE             => [
-			self::DEFAULT => '',
-		],
-		self::TEXT              => [
-			self::DEFAULT => '',
-		],
-		self::ACTION            => [
-			self::DEFAULT => '',
-		],
-	];
+	protected function defaults(): array {
+		return [
+			self::TAG     => 'div',
+			self::CLASSES => [ 'c-content-block' ],
+			self::ATTRS   => [],
+			self::LEADIN  => [],
+			self::TITLE   => [],
+			self::TEXT    => [],
+			self::ACTION  => [],
+		];
+	}
+
+	public function render(): void {
+		?>
+        <{{ tag }} {{ classes|stringify }} {{ attrs|stringify }}>
+        {{ component( 'text/Text.php', leadin ) }}
+        {{ component( 'text/Text.php', title ) }}
+        {{ component( 'text/Text.php', text ) }}
+        {{ component( 'button/Button.php', action ) }}
```

Comment:
I'm not sure if this was an intentional decision or a demonstration of what is possible. However, forcing a particular component usage within another component will be problematic. For example, we often wrap our [action (button/link) in another component (Text, for example)](https://github.com/moderntribe/square-one/pull/501/files#diff-819c8e311aa7b13860d796dce47980edR102) when the design or style warrants it.

In short, I'm not sure if these "low-level" components can support a specific child-component per usage pattern. 

Additionally, if there is no action data passed into this component, an empty button is still rendered.

Reply:
Great question! So this should entirely be viewed as simply a starting point. If you needed to change how this is output for your project, you absolutely can/should. If your project is set up in a way where this, for instance, always needs a button wrapped in another component, then you could simply update this to do so. If, on the other hand, your project needs the entire contents of this markup to be a string that you pass in already rendered and just echoed here, you could do that as well. 

I think what it'll come down to is how we think the base/default/most-used pattern should work so that it applies to the most use-cases. I'm not sure I have (or should have) a strong opinion on this, so I'll defer to whatever the FE teams thinks in regards to default setup. 

My main goal with v1 here was just to get everything rendering like it does in the old SquareOne version, and then we can pivot from there as necessary. 

This is exactly the convo I was hoping to have, so thank you for raising these questions!